### PR TITLE
CA-285268: Patching Wizard improvement: Read metadata from the update…

### PR DESCRIPTION
**Pull request by MihaelaStoica**
… file, instead of relying on the file name to match the update name in cfu

Signed-off-by: Mihaela Stoica <mihaela.stoica@citrix.com>

### CODE REVIEW

File: XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs [hunk 1]
```diff
@@ -173,10 +177,10 @@ public WizardMode WizardMode
             {
                 if (AutomatedUpdatesRadioButton.Visible && AutomatedUpdatesRadioButton.Checked)
                     return WizardMode.AutomatedUpdates;
-                var updateAlert = downloadUpdateRadioButton.Checked && dataGridViewPatches.SelectedRows.Count > 0
-                    ? (XenServerPatchAlert) ((PatchGridViewRow) dataGridViewPatches.SelectedRows[0]).UpdateAlert
+                var updateAlert = downloadUpdateRadioButton.Checked
+                    ? SelectedUpdateAlert
                     : selectFromDiskRadioButton.Checked
-                        ? GetAlertFromFileName(isValidFile(unzippedUpdateFilePath) ? unzippedUpdateFilePath.ToLowerInvariant() : fileNameTextBox.Text.ToLowerInvariant())
```
Comment (michael2012z) [inline XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs]: Are we missing the isValidFile check with the new code?
In , I didn't find any place in GetAlertFromFile() where we check file name with "isValidFile(unzippedUpdateFilePath) ? unzippedUpdateFilePath.ToLowerInvariant() : fileNameTextBox.Text.ToLowerInvariant()"

Comment (michael2012z) [reply XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs]: Sorry, I saw it. The check is in L221.
Forget it.

File: XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs [hunk 2]
```diff
@@ -276,16 +280,59 @@ protected override void PageLeaveCore(PageLoadedDirection direction, ref bool ca
...
+            }
+
+            var elements = xmlDoc.GetElementsByTagName("update");
+            var update = elements.Count > 0 ? elements[0] : null;
+
+            if (update == null || update.Attributes == null)
+                return null;
+
+            var uuid = "";
+            foreach (XmlAttribute attrib in update.Attributes)
```
[the remark this answers is not in the document]
Comment (MihaelaStoica) [reply XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs]: Great suggestion, thanks!

**Comment by michael2012z** [comment]
The latest update look good.

**Pull request merged**